packages/block-library/src/tag-cloud/style.scss

```diff
@@ -19,12 +19,16 @@
 		text-decoration: none;
 	}
 
-	&.is-style-outline a {
-		border: 1px solid currentColor;
-		font-size: unset !important; // !important Needed to override the inline styles.
-		margin-bottom: 1ch;
-		margin-right: 1ch;
-		padding: 1ch 2ch;
-		text-decoration: none !important; // !important needed to override generic post content link decoration.
+	&.is-style-outline {
+		display: flex;
+		flex-wrap: wrap;
+		gap: 1ch;
+
+		a {
```

Comment:
The [margin-right CSS from above](https://github.com/WordPress/gutenberg/pull/38995/files#diff-a8ac8407039e3829dbb76aa2d0162874e0d67df536e579e1f6167cada15722b5R13) is still being applied in addition to the gap now, since we're no longer overriding it:

<img width="1904" alt="Screen Shot 2022-02-23 at 9 06 06 AM" src="https://user-images.githubusercontent.com/5375500/155370019-5edd4724-b5b9-43f2-9944-a2bbc407e13a.png">

Would it make sense to set it to 0 for the outline style?